documentation/book/con-kafka-mirror-maker-authentication.adoc

```diff
@@ -0,0 +1,87 @@
+// Module included in the following assemblies:
+//
+// assembly-kafka-mirror-maker-authentication.adoc
+
+[id='con-kafka-mirror-maker-authentication{context}']
+= Authentication support in Kafka Mirror Maker
+
+Authentication can be configured in the `authentication` sub-property of `consumer` and `producer` properties in `KafkaMirrorMaker.spec`.
+The `authentication` property specifies the type of the authentication mechanisms which should be used and additional configuration details depending on the mechanism.
+The currently supported authentication types are:
+
+* TLS client authentication
+* SASL based authentication using SCRAM-SHA-512 mechanism
+
+
+== TLS Client Authentication
+
+To use the TLS client authentication, set the `type` property to the value `tls`.
+TLS client authentication is using TLS certificate to authenticate.
+The certificate has to be specified in the `certificateAndKey` property.
+It is always loaded from an {ProductPlatformName} secret.
+Inside the secret, it has to be stored in the X509 format under two different keys: for public and private keys.
+
+NOTE: TLS client authentication can be used only with TLS connections.
+For more details about TLS configuration in Kafka Mirror Maker see xref:assembly-kafka-mirror-maker-tls-{context}[].
+
+.An example showing TLS client authentication configuration
+[source,yaml,subs=attributes+]
+----
+apiVersion: {KafkaApiVersion}
+kind: KafkaMirrorMaker
+metadata:
+  name: my-mirror-maker
+spec:
+  # ...
+  consumer:
+    authentication:
+      type: tls
+      certificateAndKey:
+        secretName: my-source-secret
+        certificate: public.crt
+        key: private.key
+  # ...
+  producer:
+    authentication:
+      type: tls
+      certificateAndKey:
+        secretName: my-target-secret
+        certificate: public.crt
+        key: private.key
+  # ...
+----
+
+== SCRAM-SHA-512 authentication
+
+To use the authentication using the SCRAM-SHA-512 SASL mechanism, set the `type` property to the value `scram-sha-512`.
+SCRAM-SHA-512 uses a username and password to authenticate.
```

Comment:
Do we need to say that they can only use SCRAM-SHA if the broker listener they're connecting to is configured to use it?